Backport the effects of the "-svd 0" configuration flag introduced in
36.00alpha20180520 and fix a bug that prevented it from being able to
create the "out" directory on 10.6 and earlier.
https://gist.github.com/ryandesign/a7fdea794e1b84a01f15536fabe7f9ff
--- src/CNFGRAPI.h.orig	2018-06-18 10:41:10.000000000 -0500
+++ src/CNFGRAPI.h	2018-06-18 15:48:56.000000000 -0500
@@ -19,6 +19,7 @@
 
 #define RomFileName "vMac.ROM"
 #define EnableDragDrop 1
+#define SaveDialogEnable 0
 #define EnableAltKeysMode 0
 #define SwapCommandControl 0
 #define VarFullScreen 0
--- src/OSGLUCCO.m.orig	2017-07-06 21:15:00.000000000 -0500
+++ src/OSGLUCCO.m	2018-06-18 15:45:47.000000000 -0500
@@ -3960,6 +3960,67 @@
 #endif
 }
 
+#if IncludeSonyNew && ! SaveDialogEnable
+LOCALFUNC blnr FindOrMakeNamedChildDirPath(NSString *parentPath,
+	char *ChildName, NSString **childPath)
+{
+	NSString *r;
+	BOOL isDir;
+	Boolean isDirectory;
+	NSFileManager *fm = [NSFileManager defaultManager];
+	blnr v = falseblnr;
+
+	if (FindNamedChildPath(parentPath, ChildName, &r)) {
+		if ([fm fileExistsAtPath:r isDirectory: &isDir])
+		{
+			if (isDir) {
+				*childPath = r;
+				v = trueblnr;
+			} else {
+				NSString *RslvPath = MyResolveAlias(r, &isDirectory);
+				if (nil != RslvPath) {
+					if (isDirectory) {
+						*childPath = RslvPath;
+						v = trueblnr;
+					}
+				}
+			}
+		} else {
+			if ([fm respondsToSelector:@selector(
+createDirectoryAtPath:withIntermediateDirectories:attributes:error:
+				)])
+			{
+				if ([fm
+					createDirectoryAtPath:r
+					withIntermediateDirectories:NO
+					attributes:nil
+					error:nil])
+				{
+					*childPath = r;
+					v = trueblnr;
+				}
+			} else
+			if ([fm respondsToSelector:
+				@selector(createDirectoryAtPath:attributes:)])
+			{
+				if ([fm
+					createDirectoryAtPath:r
+					attributes:nil])
+				{
+					*childPath = r;
+					v = trueblnr;
+				}
+			} else
+			{
+				/* fail */
+			}
+		}
+	}
+
+	return v;
+}
+#endif
+
 @interface MyNSSavePanel : NSObject
 - (NSInteger)runModalForDirectory:(NSString *)path
 	file:(NSString *)filename;
@@ -3969,6 +4030,7 @@
 #if IncludeSonyNew
 LOCALPROC MakeNewDisk(ui5b L, NSString *drivename)
 {
+#if SaveDialogEnable
 	NSInteger result = NSCancelButton;
 	NSSavePanel *panel = [NSSavePanel savePanel];
 
@@ -4026,12 +4088,15 @@
 		NSString* filePath = [[panel URL] path];
 		MakeNewDisk0(L, filePath);
 	}
+#else /* SaveDialogEnable */
+	NSString *sPath;
 
-#if 0
-	NSString *sPath =
-		[myAppDir stringByAppendingPathComponent: drivename];
-	MakeNewDisk0(L, sPath);
-#endif
+	if (FindOrMakeNamedChildDirPath(MyDataPath, "out", &sPath)) {
+		NSString *filePath =
+			[sPath stringByAppendingPathComponent: drivename];
+		MakeNewDisk0(L, filePath);
+	}
+#endif /* SaveDialogEnable */
 }
 #endif
 
